### PR TITLE
[zero] add unit test for low-level zero init

### PR DESCRIPTION
**Pull request by 1SAA**

### CODE REVIEW

File: tests/test_zero/low_level_zero/test_zero_init.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import torch.nn as nn
+
+import colossalai
+from colossalai.tensor import ProcessGroup
+from colossalai.utils import free_port, get_current_device
+from colossalai.utils.model.colo_init_context import ColoInitContext
+from colossalai.zero import LowLevelZeroOptimizer
+
+
+class TestModel(nn.Module):
```
Comment (feifeibear) [inline tests/test_zero/low_level_zero/test_zero_init.py:16]: can we use the common model from `tests/components_to_test`?

File: tests/test_zero/low_level_zero/test_zero_init.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    assert optimizer1._dp_global_ranks == optimizer2._dp_global_ranks
+
+    mp_group1 = optimizer1._mp_torch_group
+    mp_group2 = optimizer2._mp_torch_group
+    assert dist.get_world_size(mp_group1) == dist.get_world_size(mp_group2)
+    assert dist.get_rank(mp_group1) == dist.get_rank(mp_group2)
+
+
+def run_dist(rank, world_size, port):
+    config_dict = dict(parallel=dict(tensor=dict(size=2, mode='1d')))
```
Comment (feifeibear) [inline tests/test_zero/low_level_zero/test_zero_init.py:48]: can we use tp_degree to replace 2 here?

File: tests/test_zero/low_level_zero/test_zero_init.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        self.linear2 = nn.Linear(256, 512)
+
+    def forward(self, x):
+        x = self.linear1(x)
+        x = self.linear2(x)
+        return x
+
+
+def exam_zero_init():
+    dp_2_tp_2_pg = ProcessGroup(tp_degree=2)
```
Comment (feifeibear) [inline tests/test_zero/low_level_zero/test_zero_init.py:30]: can we use tp_degree to replace 2 here?

File: tests/test_zero/low_level_zero/test_zero_init.py (hunk 1 above)
Comment (1SAA) [reply tests/test_zero/low_level_zero/test_zero_init.py:16]: it's just a initizlization test.

File: tests/test_zero/low_level_zero/test_zero_init.py (hunk 2 above)
Comment (1SAA) [reply tests/test_zero/low_level_zero/test_zero_init.py]: we just force the dp degree and tp dgree.

File: tests/test_zero/low_level_zero/test_zero_init.py (hunk 3 above)
Comment (1SAA) [reply tests/test_zero/low_level_zero/test_zero_init.py]: we just force the dp degree and tp dgree.

**Review by feifeibear — APPROVED** [review]
